projects/ontimize-web-ngx/src/lib/components/input/o-form-service-component.class.ts

```diff
@@ -334,11 +331,14 @@ export class OFormServiceComponent extends OFormDataComponent {
       });
 
       if (this._currentIndex === undefined && queryIfNotFound) {
-        if (this.queryOnBind && this.dataArray && this.dataArray.length === 0
-          && !this.cacheQueried && !this.isEmpty()) {
+        if (this.queryOnBind && this.dataArray && this.dataArray.length === 0 && !this.cacheQueried) {
```

Comment:
It would be possible to unify the two consecutive if conditions in only one? And later define the "else if ..." to make this code more readable.

Maybe you can also declare a variable to use the common value 'this.dataArray && this.dataArray.length > 0'